app/src/components/tables/ValidatorTable/ValidatorTable.test.tsx

```diff
@@ -0,0 +1,283 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { Table } from 'src/components';
+import {
+  getMockCurrentEraValidators,
+  getMockValidatorsTableFooter,
+  mockValidatorsTableColumns,
+  mockValidatorsTableOptions,
+  getMockNextEraValidators,
+  getMockValidatorsTableHeader,
+} from 'src/mocks/mock-validator-table';
+import { standardizeNumber, truncateHash } from 'src/utils';
+import { standardizePercentage } from 'src/utils/standardize-percentage';
+import userEvent from '@testing-library/user-event';
+import { render } from '../../../test-utils';
+import { EraToggleButton } from './ValidatorTable';
+
+const mockCurrentEraValidators = getMockCurrentEraValidators();
+const mockNextEraValidators = getMockNextEraValidators();
+const mockValidatorsTableHeader = getMockValidatorsTableHeader();
+const mockValidatorsTableFooter = getMockValidatorsTableFooter();
+const totalPages = Math.ceil(
+  mockCurrentEraValidators.status.validatorsCount /
+    mockValidatorsTableOptions.pagination.pageSize,
+);
+
+describe('ValidatorsTable', () => {
+  it('should render Validators Table', () => {
+    render(
+      <Table
+        header={mockValidatorsTableHeader}
+        columns={mockValidatorsTableColumns}
+        data={mockCurrentEraValidators.validators}
+        footer={mockValidatorsTableFooter}
+        tableBodyLoading={false}
+        currentPageSize={mockValidatorsTableOptions.pagination.pageSize}
+        placeholderData={{}}
+        isLastPage={
+          totalPages === mockValidatorsTableOptions.pagination.pageSize
+        }
+      />,
+    );
+
+    const validatorsTableHeader = screen.getByTestId('validators-table-header');
+    const validatorsBaseTable = screen.getByTestId('base-table');
+    const validatorsTableFooter = screen.getByTestId('validators-table-footer');
+
+    expect(validatorsTableHeader).toBeInTheDocument();
+    expect(validatorsBaseTable).toBeInTheDocument();
+    expect(validatorsTableFooter).toBeInTheDocument();
+  });
+
+  it('should render a Rank', () => {
+    render(
+      <Table
+        header={mockValidatorsTableHeader}
+        columns={mockValidatorsTableColumns}
+        data={mockCurrentEraValidators.validators}
+        footer={mockValidatorsTableFooter}
+        tableBodyLoading={false}
+        currentPageSize={mockValidatorsTableOptions.pagination.pageSize}
+        placeholderData={{}}
+        isLastPage={
+          totalPages === mockValidatorsTableOptions.pagination.pageSize
+        }
+      />,
+    );
+
+    const rank = screen.getByTestId('rank');
+    expect(rank).toHaveTextContent(
+      mockCurrentEraValidators.validators[0].rank.toString(),
+    );
+  });
+
+  it('should render a truncated Public Key', () => {
+    render(
```

Comment:
Couldn't we just render the table one time and then check for each of the column values to present after that?

Reply:
Consolidated!